src/documentation/documentationengine/documentationengine.cpp

```diff
@@ -0,0 +1,38 @@
+#include "documentationengine.h"
+
+#include <QDir>
+
+#include "documentationsearch.h"
+
+DocumentationEngine::DocumentationEngine(QObject *parent)
+    : QObject(parent)
+{
+
+}
+
+void DocumentationEngine::searchByKeyword(const QString &keyword)
+{
+    mDocumentationFiles.clear();
+    bool result{false};
+    QDir documentation{DocumentationSearch::documentationDirectory(result)};
+    if(result)
+    {
+        mDocumentationFiles = DocumentationSearch::search(documentation,keyword);
+    }
+}
```

Comment:
You use DocumentationSearch::documentationDirectory(result) on every quiry
What is probability ref will be removed during runtime?